SpatialGDKEditorCommandlet/Source/Private/Commandlets/SnapshotCommandlet.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SnapshotCommandlet.h"
+#include "SpatialGDKEditorCommandletPluginPrivate.h"
+#include "SpatialGDKEditor.h"
+#include "Runtime/Engine/Classes/Kismet/GameplayStatics.h"
+
+#include "Runtime/Engine/Classes/Engine/ObjectLibrary.h"
+#include "Runtime/Engine/Classes/Engine/World.h"
+#include "FileHelpers.h"
+
+USnapshotCommandlet::USnapshotCommandlet()
+{
+	IsClient = false;
+	IsEditor = true;
+	IsServer = false;
+	LogToConsole = true;
+}
+
+int32 USnapshotCommandlet::Main(const FString& Args)
+{
+	UE_LOG(LogSpatialGDKEditorCommandlet, Display, TEXT("Snapshot Generation Commandlet Started"));
+	UE_LOG(LogSpatialGDKEditorCommandlet, Display, TEXT("Incoming Arguments: %s"), *Args);
+
+	//NOTE: For future use, if snapshot generation configuration at the command line is desired
+	//TArray<FString> Tokens;
+	//TArray<FString> Switches;
+	//TMap<FString, FString> Params;
+	//ParseCommandLine(*Args, Tokens, Switches, Params);
+
+	GenerateSnapshots();
+
+	UE_LOG(LogSpatialGDKEditorCommandlet, Display, TEXT("Snapshot Generation Commandlet Complete"));
+
+	return 0;
+}
+
+void USnapshotCommandlet::GenerateSnapshots()
+{
+	FString MapDir = TEXT("/Game");
+	UE_LOG(LogSpatialGDKEditorCommandlet, Display, TEXT("Searching %s for maps"), *MapDir);
+	TArray<FString> MapFilePaths = GetAllMapPaths(MapDir);
+	for (FString MapFilePath : MapFilePaths)
+	{
+		GenerateSnapshotForWorld(MapFilePath);
+	}
+}
+
+void USnapshotCommandlet::GenerateSnapshotForWorld(FString WorldPath)
```

Comment:
This function seems to be creating a snapshot for a map rather than a world. Maybe rename?